### ISSUE TITLE
[REQUEST] ZGP (Zigbee Green Power) with zigpy-znp and Texas Instruments Z-Stack 3

### ISSUE THREAD

**Issue by Hedda** [zigpy/zigpy-znp#47]
@puddly Requesting support for ZGP (Zigbee Green Power) using zigpy-znp with Texas Instruments Z-Stack 3 Zigbee coordinator.

FYI, @zoic21 is trying to implement ZGP (Zigbee Green Power) device support into zigpy, however wrote that is only developing it using bellows and zigpy-deconz as he currently has no coordinator for zigpy-znp. See -> https://github.com/zigpy/zigpy/pull/519 and https://github.com/zigpy/zigpy/pull/656

PS: zoic21 is currently using EnOcean PTM 215ZE (PTM215ZE) based ZGP switch as reference but it should maybe be noted that it looks like the newer EnOcean PTM 216Z (PTM216Z) might be more common use, please see -> https://github.com/zigpy/zha-device-handlers/issues/478

PPS: There is more info about ZGP (Zigbee Green Power) and devices supporting it in the zigpy request -> https://github.com/zigpy/zigpy/issues/341

**Comment by puddly** [comment]
I can add the required endpoint but I have no way to actually test this out. Do you know of any cheaper ZGP devices? $30 for just a module is a little steep.

**Comment by Hedda** [comment]
There are links to a number of ZGP devices in https://github.com/zigpy/zigpy/issues/341 but unfortunately there looks to be no 'inexpensive' ones as of yet.

Maybe find a used one or might be possible to one on sale during next weeks Black Week and Black Friday or Cyber Monday?

Regardless it would probably be most useful to get a remote control model or a wireless switch model instead of only a module.

> Most common examples of Green Powered devices for home automation are rocker wall switches:
> 
> * https://zigbeealliance.org/zigbee_products/?product_type=certified_product&se=green+power
>   
>   * https://runlesswire.com/`click_philips_hue`/
>   * http://illumra.com/products/wireless-light-switches/zigbee-switch
>   * https://www.signify.com/global/our-company/news/press-releases/2018/20180830-new-switch-and-new-luminaire-partners-offer-more-choice-in-design-and-control-of-your-philips-hue-smart-lights
>   * https://www.cnet.com/news/this-boring-switch-might-be-the-best-philips-hue-gadget-yet/#ftag%3DCAD590a51e
>   * https://community.smartthings.com/t/theyre-here-new-green-energy-battery-free-wall-switches-for-hue-bridge-illumra-busch-jaeger-etc/135280
>   * https://zigbee.blakadder.com/Philips_9290011152X.html
>   * https://new.abb.com/low-voltage/products/residential-product/friends-of-hue-smart-switch
>   * https://www.enocean.com/en/products/enocean_modules_24ghz/ptm-216z/
>   * https://www.enocean.com/en/products/enocean_modules_24ghz/ptm-215ze/
>   * https://www.enocean.com/en/products/enocean_modules_24ghz/easyfit-single-double-rocker-wall-switch-for-zigbee-ewssz-ewsdz/
>   * https://www.enocean.com/en/products/enocean_modules_24ghz/easyfit-single-double-rocker-wall-switch-for-zigbee-ewsszg-ewsdzg/
>   * https://www.enocean.com/en/products/enocean_modules_24ghz/

**Comment by Ratief** [comment]
I would be willing to buy you a ZGP device if it would help get this working. Personally I love the Hue Taps and they are the last device that I need to migrate away from my bridge.

**Comment by Kazymire** [comment]
yes i am very interesting in zgpd too, now i using cc usb dongle with z2m, but i see that with conbbe II, we can see gpd trame. but not with cc dongle.
As Ratief, it's the last fonction i need, to migrate on zigpy
Best regard

**Comment by Hedda** [comment]
FYI, @zoic21 have made progress with ZGP for zigpy and is requesting regression testing of his new pull request in https://github.com/zigpy/zigpy/pull/656

> Can someone can test with this code if all it's ok ? No side effect. If it I think we can merge this first step (after I fix coveralls).

That is, he is asking people do functional and non-functional tests to ensure that his patch does not break any existing functions. If it does not break anything obvious in zigpy then it could possible be merged 'as-is' even if ZGP might be be fully production ready.

**Comment by sepahewe** [comment]
@puddly Would you still be helped by a device? If so, PM me on twitter (see my profile) and I'll send you one.

**Comment by omerk** [comment]
@puddly Happy to contribute to the R&D budget for this exploration, can also send a few of the CC2652R/P sticks if you need spares/test devices?

**Comment by Hedda** [comment]
> Happy to contribute to the R&D budget for this exploration, can also send a few of the CC2652R/P sticks if you need spares/test devices?

@zoic21 @sarakha63 Do you have Texas Instruments Z-Stack 3 ZNP coordinators to test with your https://github.com/zigpy/zigpy/pull/656 PR for zigpy?

**Comment by zoic21** [comment]
No we only focus for now on deconz and EZSP but after integration on this type of key we can manage integreation on ZNP.
